controller/functionApi.go

```diff
@@ -188,3 +193,62 @@ func (a *API) FunctionLogsApiPost(w http.ResponseWriter, r *http.Request) {
 	}
 	proxy.ServeHTTP(w, r)
 }
+
+// FunctionPodLogs : Get logs for a function directly from pod
+func (a *API) FunctionPodLogs(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	fnName := vars["function"]
+	ns := vars["namespace"]
+
+	if len(ns) == 0 {
+		ns = "fission-function"
+	}
+
+	f, err := a.fissionClient.Functions(api.NamespaceDefault).Get(fnName)
+	if err != nil {
+		a.respondWithError(w, err)
+		return
+	}
+	envName := f.Spec.Environment.Name
+
+	_, clientset, err := tpr.GetKubernetesClient()
+	if err != nil {
+		a.respondWithError(w, err)
+		return
+	}
+
+	// Get Unmanaged Pods first
+	selector := "unmanaged=true,functionName=" + fnName
+	podList, err := clientset.Core().Pods(ns).List(metav1.ListOptions{LabelSelector: selector})
+
+	// Get the logs for last Pod executed
+	pods := podList.Items
+	sort.Slice(pods, func(i, j int) bool {
+		itime := pods[i].ObjectMeta.CreationTimestamp.Time
+		jtime := pods[j].ObjectMeta.CreationTimestamp.Time
+		return itime.After(jtime)
+	})
+
+	podLogOpts := v1.PodLogOptions{Container: envName} // Only the env container, not fetcher
+	podLogsReq := clientset.Core().Pods(ns).GetLogs(pods[0].ObjectMeta.Name, &podLogOpts)
```

Comment:
`pods[0]` won't exist if there are no pods, and you'll get a runtime range check panic. So you have to handle the case where there are zero pods.